src/AggregateFunctions/AggregateFunctionQuantileExactWeighted.cpp

```diff
@@ -197,23 +232,167 @@ struct QuantileExactWeighted
         }
     }
 
-    /// The same, but in the case of an empty state, NaN is returned.
-    Float64 getFloat(Float64) const
+    /// getFloat implementation without interpolation
+    Float64 getFloatImpl(Float64) const
     {
         throw Exception(ErrorCodes::NOT_IMPLEMENTED, "Method getFloat is not implemented for QuantileExact");
     }
 
-    void getManyFloat(const Float64 *, const size_t *, size_t, Float64 *) const
+    /// getManyFloat implementation without interpolation
+    void getManyFloatImpl(const Float64 *, const size_t *, size_t, Float64 *) const
     {
         throw Exception(ErrorCodes::NOT_IMPLEMENTED, "Method getManyFloat is not implemented for QuantileExact");
     }
-};
 
+    /// get implementation with interpolation
+    Value getInterpolatedImpl(Float64 level) const
+    {
+        size_t size = map.size();
+        if (0 == size)
+            return std::numeric_limits<Value>::quiet_NaN();
```

Comment:
This is odd (in both cases). `quantileExactWeighted` returns 0 in this case. I would prefer to keep it consistent.

Reply:
fixed.